packages/core/src/utils/assertDuplicates.ts

```diff
@@ -0,0 +1,23 @@
+import type { ProofRequest } from '../modules/proofs/models/ProofRequest'
+
+import { AriesFrameworkError } from '../error/AriesFrameworkError'
+
+import { attributesToArray, predicatesToArray } from './toArray'
+
+export function checkProofRequestForDuplicates(proofRequest: ProofRequest) {
+  const attributes = attributesToArray(proofRequest)
+  const predicates = predicatesToArray(proofRequest)
+  assertDuplicatesInArray(attributes)
+  assertDuplicatesInArray(predicates)
+  assertDuplicatesInArray(attributes.concat(predicates))
+}
+
+export function assertDuplicatesInArray(arr: string[]) {
```

Comment:
The name of this method makes me assume we want to assert there are duplicates in the array. While we want to make sure there are not duplicates in the array

```suggestion
export function assertNoDuplicatesInArray(array: string[]) {
```